packages/order-utils/src/call_data_utils.ts

```diff
@@ -0,0 +1,86 @@
+import ethUtil = require('ethereumjs-util');
+import * as ethers from 'ethers';
+import * as _ from 'lodash';
+
+import { constants } from './constants';
+
+export type MethodAbi = ethers.utils.FunctionFragment;
+
+export interface CallParams {
+    [paramName: string]: any;
+}
+
+export interface DecodedCallData {
+    name: string;
+    callParams: CallParams;
+}
+
+export interface AbiBySelector {
+    [selector: string]: MethodAbi;
+}
+/**
+ * Converts the list of ABIs to the hash indexed by the 4-byte selector.
+ */
+export function abisToAbiBySelector(abis: MethodAbi[]): AbiBySelector {
+    const abiBySelector = _.reduce<MethodAbi, AbiBySelector>(
+        abis,
+        (abiBySelectorAccumulator: AbiBySelector, methodAbi: MethodAbi) => {
+            const selector = getMethodSelector(methodAbi);
+            return {
+                ...abiBySelectorAccumulator,
+                [selector]: methodAbi,
+            };
+        },
+        {},
+    );
+    return abiBySelector;
+}
+
+/**
+ * Computes the 4-byte selector of the method.
+ * @param methodAbi Method ABI
+ */
+export function getMethodSelector(methodAbi: MethodAbi): string {
```

Comment:
This should be pushed upstream to ethersjs.

Reply:
Will do

Reply:
Removed. Using `AbiEncoder` now